tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/SdkBuildTasks/ArchiveSymbolsTask.cs

```diff
@@ -0,0 +1,522 @@
+        const string OUTPUT_ROOTDIRNAME_DEBUG = "Debug";
+        const string OUTPUT_ROOTDIRNAME_RELEASE = "Release";
+        #endregion
+
+        #region fields
+        private string _symbolsArchiveRootDir;
+        private string _buildJobId;
+        private List<string> _assemblyList;
+        #endregion
+
+        protected override INetSdkTask TaskInstance => this;
+
+        /// <summary>
+        /// List of ITaskItem that represents built assembly files        
+        /// </summary>
+        [Required]
+        public ITaskItem[] BuiltAssemblyFileCollection { get; set; }
+
+        /// <summary>
+        /// UNC path where symbols/assemblies will be copied which is used by the Symbols Archive process that will process your archive requests at a later stage
+        /// </summary>
+        [Required]
+        public string ArchiveSymbolsRootDir
+        {
+            get
+            {
+                if (!Directory.Exists(_symbolsArchiveRootDir))
+                {
+                    string alternatePath = System.Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
+                    if (!File.Exists(alternatePath))
+                    {
+                        string expMsg = string.Format("Provided path '{0}', does not exists and unable to determine alternate Symbols Archive directory path.", _symbolsArchiveRootDir);
+                        throw new ApplicationException(expMsg);
+                    }
+                }
+
+                return _symbolsArchiveRootDir;
+            }
+            set
+            {
+                _symbolsArchiveRootDir = value;
+            }
+        }
+
+        /// <summary>
+        /// List of symbol archive request files
+        /// This is primarily used for test purpose
+        /// </summary>
+        [Output]
+        public List<string> SymbolsRequestFileList { get; set; }
+
+
+        /// <summary>
+        /// In case the process of archiving symbols fail, this email will be used to communicate the current status of your archive request
+        /// </summary>
+        public string ArchiveSymbolsRequestStatusEmail { get; set; }
+
+        /// <summary>
+        /// This is an internal list that is used to process assembly list
+        /// Rather than dealing with ITaskItems, this is a list of ITaskItem.ItemSpec
+        /// </summary>
+        private List<string> BuiltAssemblyList
+        {
+            get
+            {
+                if(_assemblyList == null)
+                {
+                    if(BuiltAssemblyFileCollection != null)
+                    {
+                        if(BuiltAssemblyFileCollection.Any<ITaskItem>())
+                        {
+                            _assemblyList = BuiltAssemblyFileCollection.Select<ITaskItem, string>((item) => item.ItemSpec)?.ToList<string>();
+                        }
+                        else
+                        {
+                            _assemblyList = new List<string>();
```

Comment:
Shouldn't this be it the outer else statement to ensure you never return null from this property?

Reply:
Good catch. Will move up to avoid null.